engines/private/funcs.cpp

```diff
@@ -0,0 +1,784 @@
+        // in the original scripts
+        if (g_private->_playedPhoneClips.contains(*sound))
+            return;
+
+        g_private->_playedPhoneClips.setVal(*sound, true);
+        PhoneInfo *p = (PhoneInfo *)malloc(sizeof(PhoneInfo));
+        p->sound = sound;
+        p->flag = flag;
+        p->val = val;
+        g_private->_phone.push_back(*p);
+    }
+
+    else
+        error("error: invalid sound type %s", t);
+}
+
+void fAMRadioClip(ArgArray args) {
+    assert(args.size() <= 4);
+    fAddSound(args[0].u.str, "AMRadioClip");
+}
+
+void fPoliceClip(ArgArray args) {
+    assert(args.size() <= 4);
+    fAddSound(args[0].u.str, "PoliceClip");
+}
+
+void fPhoneClip(ArgArray args) {
+    if (args.size() == 2) {
+        debugC(1, kPrivateDebugScript, "Unimplemented PhoneClip special case");
+        return;
+    }
+    int i = args[2].u.val;
+    int j = args[3].u.val;
+
+    if (i == j)
+        fAddSound(args[0].u.str, "PhoneClip", args[4].u.sym, args[5].u.val);
+    else {
+        assert(i < j);
+        char *clip = g_private->getRandomPhoneClip(args[0].u.str, i, j);
+        fAddSound(clip, "PhoneClip", args[4].u.sym, args[5].u.val);
+    }
+}
+
+void fSoundArea(ArgArray args) {
+    // assert types
+    char *n;
+    if (args[1].type == NAME)
+        n = (char *)args[1].u.sym->name->c_str();
+    else if (args[1].type == STRING)
+        n = args[1].u.str;
+    else
+        assert(0);
+
+    debugC(1, kPrivateDebugScript, "SoundArea(%s, %s)", args[0].u.str, n);
+    if (strcmp(n, "kAMRadio") == 0) {
+        Common::String *s = new Common::String(args[0].u.str);
+        MaskInfo *m = (MaskInfo *)malloc(sizeof(MaskInfo));
+        m->surf = g_private->loadMask(*s, 0, 0, true);
+        m->cursor = args[2].u.sym->name;
+        m->nextSetting = NULL;
+        m->flag1 = NULL;
+        m->flag2 = NULL;
+        g_private->_AMRadioArea = m;
+        g_private->_masks.push_front(*m);
+    } else if (strcmp(n, "kPoliceRadio") == 0) {
+        Common::String *s = new Common::String(args[0].u.str);
+        MaskInfo *m = (MaskInfo *)malloc(sizeof(MaskInfo));
+        m->surf = g_private->loadMask(*s, 0, 0, true);
+        m->cursor = args[2].u.sym->name;
+        m->nextSetting = NULL;
+        m->flag1 = NULL;
+        m->flag2 = NULL;
+        g_private->_policeRadioArea = m;
+        g_private->_masks.push_front(*m);
+    } else if (strcmp(n, "kPhone") == 0) {
+        Common::String *s = new Common::String(args[0].u.str);
+        MaskInfo *m = (MaskInfo *)malloc(sizeof(MaskInfo));
+        m->surf = g_private->loadMask(*s, 0, 0, true);
+        m->cursor = args[2].u.sym->name;
+        m->nextSetting = NULL;
+        m->flag1 = NULL;
+        m->flag2 = NULL;
+        g_private->_phoneArea = m;
+        g_private->_masks.push_front(*m);
+    }
+}
+
+void fSafeDigit(ArgArray args) {
+    debugC(1, kPrivateDebugScript, "WARNING: SafeDigit is not implemented");
```

Comment:
Why not use `warning()` directly?

Reply:
It won't allow to select the proper debug channel (`kPrivateDebugScript`)